### PR TITLE
[Sealing] New SoR for sealing & verification

### PR DESCRIPTION
**Pull request by durkmurder**
(dapperlabs/flow-go/issues/5551)[https://github.com/dapperlabs/flow-go/issues/5551]

### Context

During sealing & verification phase 2 we were using SoR(source of randomness) from block that was executed. For phase 3 we need to change that SoR will be picked at block that incorporates execution result.

### Changes in this PR

- Located all places where incorporated result is created and changed `result.blockID` -> `incorporatedBlockID`
- Adjusted logic both for verification and consensus nodes
- Adjusted tests
- Adjusted docs

### CODE REVIEW

**Review by zhangchiqing — COMMENTED** [review]
Changes look good. Be careful to merge, as we haven't launched Phase 2 yet, merge and deploy would jump from phase 1 to phase 3.  I would suggest to merge into a feature branch

File: engine/verification/utils/unittest/helper.go
Comment (yhassanzadeh13) [inline engine/verification/utils/unittest/helper.go:340]: 👍

**Review by yhassanzadeh13 — APPROVED** [review]
From the verification, the node perspective looks good, thanks for doing the refactor. I don't have a strong context about the consensus part but tried my best to review that part as well. Hence, please rely on the consensus experts' reviews (@AlexHentschel and @zhangchiqing) for the consensus domain changes. 

Just want to highlight that the chunk assignment discrepancy can lead to unnoticed breaks in the sealing and verification pipeline, I would recommend please run  https://github.com/onflow/flow-go/pull/922 on your branch (locally) and make sure that `verification/TestSealingHappyPath` is going through. That PR is currently blocked by another PR but will go to master hopefully soon. So in case, there is no rush for the current PR, you may block it by https://github.com/onflow/flow-go/pull/922. I leave it up to you.

**Comment by durkmurder** [comment]
> Changes look good. Be careful to merge, as we haven't launched Phase 2 yet, merge and deploy would jump from phase 1 to phase 3. I would suggest to merge into a feature branch

I believe we are good to go with phase 3. Phase 2 is completed at this point and we can start deploying phase 3 on Canary.
@AlexHentschel

**Comment by durkmurder** [comment]
> From the verification, the node perspective looks good, thanks for doing the refactor. I don't have a strong context about the consensus part but tried my best to review that part as well. Hence, please rely on the consensus experts' reviews (@AlexHentschel and @zhangchiqing) for the consensus domain changes.
> 
> Just want to highlight that the chunk assignment discrepancy can lead to unnoticed breaks in the sealing and verification pipeline, I would recommend please run #922 on your branch (locally) and make sure that `verification/TestSealingHappyPath` is going through. That PR is currently blocked by another PR but will go to master hopefully soon. So in case, there is no rush for the current PR, you may block it by #922. I leave it up to you.

@yhassanzadeh13 I have troubles running integration tests locally, instead I have run this branch on benchnet and everything works as expected. If you think this is good enough then we can merge it and wait for your tests then.

**Comment by yhassanzadeh13** [comment]
> @yhassanzadeh13 I have troubles running integration tests locally, instead I have run this branch on benchnet and everything works as expected. If you think this is good enough then we can merge it and wait for your tests then.

@durkmurder my integration test has just been merged on master, please feel free to merge your PR.

**Review by zhangchiqing — APPROVED** [review]
🚀

File: module/builder/consensus/builder.go
Comment (AlexHentschel) [inline module/builder/consensus/builder.go:399]: I think we can remove this comment
```golang
// ATTENTION:
// Here, IncorporatedBlockID (the first argument) should be set to
// ancestorID, because that is the block that contains the
// ExecutionResult.
```

File: module/builder/consensus/builder.go
Comment (AlexHentschel) [inline module/builder/consensus/builder.go:387]: [optional suggestion for performance improvements]
using `b.index` and the retrieving the results individually is more efficient that assembling the entire block payload

File: module/validation/seal_validator.go
Comment (AlexHentschel) [inline module/validation/seal_validator.go:176]: I think we can remove this comment entirely
```golang
// ATTENTION:
// Here, IncorporatedBlockID (the first argument) should be set
// to ancestorID, because that is the block that contains the
// ExecutionResult.
```

**Review by AlexHentschel — APPROVED** [review]
🎸 

I have added a couple minor suggestions in this PR: https://github.com/onflow/flow-go/pull/1070

**Pull request merged**
